code_samples/api/rest_api/create_image.xml.php

```diff
@@ -66,94 +71,68 @@
 
-$response = curl_exec($curl);
-
-if ($error = curl_error($curl)) {
-    echo "cURL error: $error\n";
-    curl_close($curl);
+try {
+    $response = $client->request('POST', "$baseUrl/content/objects", [
+        'headers' => [
+            'Content-Type: application/vnd.ibexa.api.ContentCreate+xml',
+            'Accept: application/vnd.ibexa.api.ContentInfo+xml',
+        ],
+        'body' => $data,
+    ]);
+} catch (HttpException\TransportExceptionInterface $exception) {
+    echo "Client error: {$exception->getMessage()}\n";
     exit(3);
 }
 
-if (201 !== $responseCode = curl_getinfo($curl, CURLINFO_RESPONSE_CODE)) {
-    if (!empty($response) && $doc->loadXML($response) && 'ErrorMessage' === $doc->firstChild->nodeName) {
+if (201 !== $responseCode = $response->getStatusCode()) {
+    if (!empty($response->getContent(false)) && @$doc->loadXML($response->getContent(false)) && 'ErrorMessage' === $doc->firstChild->nodeName) {
         echo "Server error: {$doc->getElementsByTagName('errorCode')->item(0)->nodeValue} {$doc->getElementsByTagName('errorMessage')->item(0)->nodeValue}\n";
         echo "\t{$doc->getElementsByTagName('errorDescription')->item(0)->nodeValue}\n";
-        curl_close($curl);
         exit(4);
     }
+    $responseHeaders = $response->getInfo('response_headers');
     $error = $responseHeaders[0] ?? $responseCode;
     echo "Server error: $error\n";
-    curl_close($curl);
     exit(5);
 }
 
-$doc->loadXML($response);
+$doc->loadXML($response->getContent());
 
 if ('Content' !== $doc->firstChild->nodeName || !$doc->firstChild->hasAttribute('id')) {
     echo "Response error: Unexpected response structure\n";
-    curl_close($curl);
     exit(6);
 }
 
 $contentId = $doc->firstChild->getAttribute('id');
 
-curl_reset($curl);
-$responseHeaders = [];
-
-curl_setopt_array($curl, [
-    CURLOPT_USERPWD => "$username:$password",
-    CURLOPT_URL => "$baseUrl/content/objects/$contentId/versions/1",
-    CURLOPT_POST => true,
-    CURLOPT_HTTPHEADER => [
-        'X-HTTP-Method-Override: PUBLISH',
-        'Accept: application/json',
-    ],
-    CURLOPT_RETURNTRANSFER => true,
-]);
-
-$response = curl_exec($curl);
-
-if ($error = curl_error($curl)) {
-    echo "cURL error: $error\n";
-    curl_close($curl);
+try {
+    $response = $client->request('PUBLISH', "$baseUrl/content/objects/$contentId/versions/1", [
+        'headers' => [
+            'Accept: application/xml',
+        ],
+    ]);
+} catch (HttpException\TransportExceptionInterface $exception) {
+    echo "Client error: {$exception->getMessage()}\n";
     exit(7);
 }
 
-if (204 !== $responseCode = curl_getinfo($curl, CURLINFO_RESPONSE_CODE)) {
-    if (!empty($response) && $doc->loadXML($response) && 'ErrorMessage' === $doc->firstChild->nodeName) {
+if (204 !== $responseCode = $response->getStatusCode()) {
+    if (!empty($response->getContent(false)) && @$doc->loadXML($response->getContent(false)) && 'ErrorMessage' === $doc->firstChild->nodeName) {
```

Comment:
I don't like silencing errors from XML:
```php
@$doc->loadXML($response->getContent(false))
```

It's considered a bad practice. I think in this example `@` can be omitted. I don't think we want to provide full validation in this example, as there would be much more failure points.

Reply:
The only case I found of not returning proper XML is on the rare "500 Internal Server Error". No need to hide all those "Warning: DOMDocument::loadXML(): Entity 'nbsp' not defined in Entity" "Opening and ending tag mismatch" due to HTML being returned. I'll remove them.

Reply:
Suppressor unscrewed in 4a720e83e12115167d86975f556a9e237a98352a